plugins/woocommerce-blocks/assets/js/data/cart/reducers.ts

```diff
@@ -80,8 +80,20 @@ const reducer: Reducer< CartState > = (
 			}
 			break;
 		case types.SET_BILLING_ADDRESS:
+			const billingAddressChanged = Object.keys(
+				action.billingAddress
+			).some( ( key ) => {
+				return (
+					action.billingAddress[ key ] !==
+					state.cartData.billingAddress?.[ key ]
+				);
+			} );
 			state = {
 				...state,
+				metaData: {
+					...state.metaData,
+					isCustomerDataDirty: billingAddressChanged,
```

Comment:
Potential logic bug; if this is true, but this particular dispatch did not result in a change, a `true` value will switch to `false`.

Reply:
Hm, yes you're right. Probably should be
```suggestion
					isCustomerDataDirty: state.metaData.isCustomerDataDirty || billingAddressChanged,
```

suppose this also adds to the case for timestamp-based tracking, though.